test/generation.test.js

```diff
@@ -11,13 +13,101 @@ function extractLocalesFromAcceptLanguageHeader(acceptLanguageHeader) {
     return extractedLocales;
 }
 
+const getUserAgent = (headers) => {
+    let userAgent;
+    for (const [header, value] of Object.entries(headers)) {
+        if (header.toLowerCase() === 'user-agent') {
+            userAgent = value;
+            break;
+        }
+    }
+
+    return userAgent;
+};
+
+const getBrowser = (userAgent) => {
+    let browser;
+    if (userAgent.includes('Firefox')) {
+        browser = 'firefox';
+    } else if (userAgent.includes('Chrome')) {
+        browser = 'chrome';
+    } else {
+        browser = 'safari';
+    }
+
+    return browser;
+};
+
 describe('Generation tests', () => {
     const headerGenerator = new HeaderGenerator({
         httpVersion: '2',
     });
 
     test('Generates headers', () => {
-        expect(headerGenerator.getHeaders()).toBeTruthy();
+        const headers = headerGenerator.getHeaders();
+        const userAgent = getUserAgent(headers);
+        const browser = getBrowser(userAgent);
+
+        expect(typeof headers).toBe('object');
+
+        const order = headersOrder[browser];
+
+        let index = -1;
+        for (const header of Object.keys(headers)) {
```

Comment:
Maybe you should add the comments from the resolved conversations around this one to the actual code, this test isn't trivial to understand :)

Reply:
sure thing :)